src/plugins/data/public/index_patterns/fields/field.test.ts

```diff
@@ -0,0 +1,223 @@
+    expect(field.subType).toEqual(fieldValues.subType);
+
+    // not writable, not serialized
+    expect(() => {
+      field.indexPattern = {} as IndexPattern;
+    }).toThrow();
+
+    // computed fields
+    expect(() => {
+      field.format = { name: 'newFormatName' };
+    }).toThrow();
+
+    expect(() => {
+      field.sortable = false;
+    }).toThrow();
+
+    expect(() => {
+      field.filterable = false;
+    }).toThrow();
+
+    expect(() => {
+      field.visualizable = false;
+    }).toThrow();
+
+    expect(() => {
+      field.displayName = 'newDisplayName';
+    }).toThrow();
+
+    expect(() => {
+      field.$$spec = { a: 'b' };
+    }).toThrow();
+  });
+
+  it('sets type field when _source field', () => {
+    const field = getField({ name: '_source' });
+    expect(field.type).toEqual('_source');
+  });
+
+  it('calculates seaarchable', () => {
+    const field = getField({ searchable: true, scripted: false });
+    expect(field.searchable).toEqual(true);
+
+    const fieldB = getField({ searchable: false, scripted: true });
+    expect(fieldB.searchable).toEqual(true);
+
+    const fieldC = getField({ searchable: false, scripted: false });
+    expect(fieldC.searchable).toEqual(false);
+  });
+
+  it('calculates aggregatable', () => {
+    const field = getField({ aggregatable: true, scripted: false });
+    expect(field.aggregatable).toEqual(true);
+
+    const fieldB = getField({ aggregatable: false, scripted: true });
+    expect(fieldB.aggregatable).toEqual(true);
+
+    const fieldC = getField({ aggregatable: false, scripted: false });
+    expect(fieldC.aggregatable).toEqual(false);
+  });
+
+  it('calculates readFromDocValues', () => {
+    const field = getField({ readFromDocValues: true, scripted: false });
+    expect(field.readFromDocValues).toEqual(true);
+
+    const fieldB = getField({ readFromDocValues: false, scripted: false });
+    expect(fieldB.readFromDocValues).toEqual(false);
+
+    const fieldC = getField({ readFromDocValues: true, scripted: true });
+    expect(fieldC.readFromDocValues).toEqual(false);
+  });
+
+  it('calculates sortable', () => {
+    const field = getField({ name: '_score' });
+    expect(field.sortable).toEqual(true);
+
+    const fieldB = getField({ indexed: true, type: KBN_FIELD_TYPES.STRING });
+    expect(fieldB.sortable).toEqual(true);
+
+    const fieldC = getField({ indexed: false });
+    expect(fieldC.sortable).toEqual(false);
+  });
+
+  it('calculates filterable', () => {
+    const field = getField({ name: '_id' });
+    expect(field.filterable).toEqual(true);
+
+    const fieldB = getField({ scripted: true });
+    expect(fieldB.filterable).toEqual(true);
+
+    const fieldC = getField({ indexed: true, type: KBN_FIELD_TYPES.STRING });
+    expect(fieldC.filterable).toEqual(true);
+
+    const fieldD = getField({ scripted: false, indexed: false });
+    expect(fieldD.filterable).toEqual(false);
+  });
+
+  it('exports the property to JSON', function () {
```

Comment:
consistency nit - some of these tests are using `function ()` and some are using `() =>`